katello/katello-change-hostname

```diff
@@ -0,0 +1,174 @@
+    hostname = `hostname`
+  else
+    hostname = Socket.gethostname
+  end
+  hostname.chomp 
+end
+
+options = {}
+options[:program] = DEFAULT_PROGRAM
+options[:scenario] = DEFAULT_SCENARIO
+options[:ssl_dir] = DEFAULT_SSL_DIR
+
+opt_parser = OptionParser.new do |opt|
+  opt.banner = "Usage: katello-change-hostname HOSTNAME [OPTIONS]"
+  opt.separator  ""
+  opt.separator  "Example:"
+  opt.separator  " katello-change-hostname foo.example.com -u admin -p changeme"
+  opt.separator  ""
+  opt.separator  "Options"
+
+  opt.on("-u","--username USERNAME","admin username REQUIRED") do |username|
+    options[:username] = username
+  end
+
+  opt.on("-p","--password PASSWORD","admin password REQUIRED") do |password|
+    options[:password] = password
+  end
+
+  opt.on("-g","--program PROGRAM","name of the program you are modifying (defaults to #{DEFAULT_PROGRAM})") do |program|
+    options[:program] = program
+  end
+
+  opt.on("-s","--scenario SCENARIO","name of the scenario you are modifying (defaults to #{DEFAULT_SCENARIO})") do |scenario|
+    options[:scenario] = scenario
+  end
+
+  opt.on("-d","--disable-system-checks","runs the installer with --disable-system-checks") do |system_check|
+    options[:system_check] = true
+  end
+
+  opt.on("-l","--ssl-dir SSL_DIR","specify the ssl cert directory to replace (defaults to #{DEFAULT_SSL_DIR})") do |ssl_dir|
+    options[:ssl_dir] = ssl_dir
+  end
+
+  opt.on("-h","--help","help") do
+    puts opt_parser
+    exit
+  end
+end
+opt_parser.parse!
+
+raise 'Must run as root' unless Process.uid == 0
+
+unless options[:username] && options[:password]
+  STDOUT.puts "Username and/or Password options are missing!"
+  puts opt_parser
+  exit
+end
+
+if options[:scenario] == "katello"
+  proxy = "Smart Proxy"
+  plural_proxy = "Smart Proxies"
+else
+  proxy = "Capsule"
+  plural_proxy = "Capsules"
+end
+
+if ARGV[0] && ARGV.count >= 1
+  new_hostname = ARGV[0]
+else
+  puts opt_parser
+  exit
+end
+
+# Get the hostname from your system
+old_hostname = get_hostname
+
+STDOUT.puts "Updating default #{proxy}"
+default_capsule_id = `hammer -u #{options[:username]} -p #{options[:password]} capsule list | grep #{old_hostname} | awk '{ print $1 }'`[0]
+capsule_hammer = `hammer -u #{options[:username]} -p #{options[:password]} capsule update --id #{default_capsule_id} --url https://#{new_hostname}:9090 --new-name #{new_hostname}`
+
+STDOUT.puts "Updating installation media paths"
+installation_media_ids = `hammer -u #{options[:username]} -p #{options[:password]} medium list | grep "#{old_hostname}" | awk '{ print $1 }'`.split("\n")
+installation_media_paths = `hammer -u #{options[:username]} -p #{options[:password]} medium list | grep "#{old_hostname}" | awk '{ print $5 }'`.split("\n")
+installation_media_ids.zip(installation_media_paths).each do |id, path|
+  `hammer medium update -u #{options[:username]} -p #{options[:password]} --id #{id} --path #{path.sub! old_hostname, new_hostname}`   
```

Comment:
@johnpmitsch Here's the offending line where the options are after the subcommand.